geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+# GEP-1713: Standard Mechanism to Merge Multiple Gateways
+
+* Issue: [#1713](/kubernetes-sigs/gateway-api/issues/1713)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## tl;dr
+
+The `Gateway` Resource is a point of contention since it is the only place to attach listeners with certificates. We propose a new resource called `ListenerSet` to allow a shared list of listeners to be attached to a single `Gateway`.
+
+## Goals
+- Define a mechanism to merge listeners into a single `Gateway`
+
+## Future Goals (Beyond the GEP)
+- Attaching listeners to `Gateways` in different namespaces
+
+## Introduction
+
+Knative generates on demand per-service certificates using HTTP-01 challenges. 
+There can be O(1000) Knative `Services` in the cluster which means we have O(1000) distinct certificates. 
+Thus updating a single `Gateway` resource with this many certificates is a contention point and inhibits horizontal scaling of our controllers.
+
+More broadly, large scale gateway users often expose `O(1000)` domains, but are currently limited by the maximum of 64 `listeners`.
+
+The spec currently has language to indicate implementations `MAY` merge `Gateways` resources but the mechanic isn't defined.
+https://github.com/kubernetes-sigs/gateway-api/blob/541e9fc2b3c2f62915cb58dc0ee5e43e4096b3e2/apis/v1beta1/gateway_types.go#L76-L78
+
+## API
```

Comment:
Before this section, we need to define a feature name that we can use to talk about this later.

I propose `ListenerSet`, so that we can have sub-features like `ListenerSetSubFeature` as feature names for conformance.

The ListenerSet feature will need to be added to the _experimental_ branches as an _extended_ feature, which implementations can choose to support. At that time, all the requirements in this document that use MUST apply to their implementation of the feature.

This is important later because we need to minimize the changes to Gateway while we experiment with this idea.

Reply:
done